packages/fxa-auth-server/test/local/push.js

```diff
@@ -70,110 +78,69 @@ describe('push', () => {
         pushEndpointExpired: false,
       },
     ];
+    mockStatsD = {
+      increment: sinon.spy()
+    };
+    mockSendNotification = sinon.spy(async () => {});
   });
 
-  it('sendPush does not reject on empty device array', () => {
-    const thisMockLog = mockLog({
-      info: function (op, log) {
-        if (log.name === 'push.account_verify.success') {
-          assert.fail('must not call push.success');
-        }
-      },
-    });
-    const push = require(pushModulePath)(thisMockLog, mockDb, mockConfig);
-
-    return push.sendPush(mockUid, [], 'accountVerify');
+  it('sendPush does not reject on empty device array', async () => {
+    const push = loadMockedPushModule();
+    await push.sendPush(mockUid, [], 'accountVerify');
+    assert.equal(mockSendNotification.callCount, 0);
+    assert.equal(mockLog.info.callCount, 0);
+    assert.equal(mockStatsD.increment.callCount, 0);
   });
 
-  it('sendPush sends notifications with a TTL of 0', () => {
-    let successCalled = 0;
-    const thisMockLog = mockLog({
-      info: function (op, log) {
-        if (log.name === 'push.account_verify.success') {
-          // notification sent
-          successCalled++;
-        }
-      },
-    });
-
-    const mocks = {
-      'web-push': {
-        sendNotification: function (sub, payload, options) {
-          assert.equal(options.TTL, '0', 'sends the proper ttl header');
-          return P.resolve();
-        },
-      },
-    };
-
-    const push = proxyquire(pushModulePath, mocks)(
-      thisMockLog,
-      mockDb,
-      mockConfig
-    );
-    return push.sendPush(mockUid, mockDevices, 'accountVerify').then(() => {
-      assert.equal(successCalled, 2);
-    });
+  it('sendPush logs metrics about successful sends', async () => {
+    const push = loadMockedPushModule();
+    await push.sendPush(mockUid, mockDevices, 'accountVerify');
+    assert.equal(mockSendNotification.callCount, 2);
+    assert.equal(mockLog.info.callCount, 5);
+    assert.equal(mockLog.info.getCall(0).args[0], "push.filteredUnsupportedDevice")
+    assert.equal(mockLog.info.getCall(1).args[0], "push.send.attempt")
+    assert.equal(mockLog.info.getCall(1).args[1].reason, "accountVerify")
+    assert.equal(mockLog.info.getCall(1).args[1].deviceId, mockDevices[0].id)
+    assert.equal(mockLog.info.getCall(2).args[0], "push.send.success")
+    assert.equal(mockLog.info.getCall(2).args[1].reason, "accountVerify")
+    assert.equal(mockLog.info.getCall(2).args[1].deviceId, mockDevices[0].id)
+    assert.equal(mockLog.info.getCall(3).args[0], "push.send.attempt")
+    assert.equal(mockLog.info.getCall(3).args[1].reason, "accountVerify")
+    assert.equal(mockLog.info.getCall(3).args[1].deviceId, mockDevices[1].id)
+    assert.equal(mockLog.info.getCall(4).args[0], "push.send.success")
+    assert.equal(mockLog.info.getCall(4).args[1].reason, "accountVerify")
+    assert.equal(mockLog.info.getCall(4).args[1].deviceId, mockDevices[1].id)
```

Comment:
@mozilla/fxa-devs before I get too far down the path of asyncifying and cleaning up these tests, do y'all have a current preferred style for doing this sort of "assert that calls were made as expected" stuff using sinon? I find the above pretty wordy, but clearer and more robust than what it was doing before.

Reply:
You could try using sinon.assert.  For example, `sinon.assert.calledWithExactly(mockLock, ...)`.